lib/icinga/downtime.cpp

```diff
@@ -16,7 +16,7 @@ using namespace icinga;
 
 static int l_NextDowntimeID = 1;
 static std::mutex l_DowntimeMutex;
-static std::map<int, String> l_LegacyDowntimesCache;
+static std::map<int, Downtime::Ptr> l_LegacyDowntimesCache;
```

Comment:
Oh, we accidentally stumbled upon a memory leak: entries are only inserted into but never removed from `l_LegacyDowntimesCache`. Previously, it would "only" leak the downtime name, with this change it will leak the whole downtime object.